pallets/dapp-staking-v3/src/lib.rs

```diff
@@ -0,0 +1,2028 @@
+                        .as_mut()
+                        .ok_or(Error::<T>::ContractNotFound)?;
+
+                    ensure!(dapp_info.owner == dev_account, Error::<T>::OriginNotOwner);
+
+                    dapp_info.reward_destination = beneficiary.clone();
+
+                    Ok(())
+                },
+            )?;
+
+            Self::deposit_event(Event::<T>::DAppRewardDestinationUpdated {
+                smart_contract,
+                beneficiary,
+            });
+
+            Ok(())
+        }
+
+        /// Used to change dApp owner.
+        ///
+        /// Can be called by dApp owner or dApp staking manager origin.
+        /// This is useful in two cases:
+        /// 1. when the dApp owner account is compromised, manager can change the owner to a new account
+        /// 2. if project wants to transfer ownership to a new account (DAO, multisig, etc.).
+        #[pallet::call_index(3)]
+        #[pallet::weight(T::WeightInfo::set_dapp_owner())]
+        pub fn set_dapp_owner(
+            origin: OriginFor<T>,
+            smart_contract: T::SmartContract,
+            new_owner: T::AccountId,
+        ) -> DispatchResult {
+            Self::ensure_pallet_enabled()?;
+            let origin = Self::ensure_signed_or_manager(origin)?;
+
+            IntegratedDApps::<T>::try_mutate(
+                &smart_contract,
+                |maybe_dapp_info| -> DispatchResult {
+                    let dapp_info = maybe_dapp_info
+                        .as_mut()
+                        .ok_or(Error::<T>::ContractNotFound)?;
+
+                    // If manager origin, `None`, no need to check if caller is the owner.
+                    if let Some(caller) = origin {
+                        ensure!(dapp_info.owner == caller, Error::<T>::OriginNotOwner);
+                    }
+
+                    dapp_info.owner = new_owner.clone();
+
+                    Ok(())
+                },
+            )?;
+
+            Self::deposit_event(Event::<T>::DAppOwnerChanged {
+                smart_contract,
+                new_owner,
+            });
+
+            Ok(())
+        }
+
+        /// Unregister dApp from dApp staking protocol, making it ineligible for future rewards.
+        /// This doesn't remove the dApp completely from the system just yet, but it can no longer be used for staking.
+        ///
+        /// Can be called by dApp staking manager origin.
+        #[pallet::call_index(4)]
+        #[pallet::weight(T::WeightInfo::unregister())]
+        pub fn unregister(
+            origin: OriginFor<T>,
+            smart_contract: T::SmartContract,
+        ) -> DispatchResult {
+            Self::ensure_pallet_enabled()?;
+            T::ManagerOrigin::ensure_origin(origin)?;
+
+            let current_era = ActiveProtocolState::<T>::get().era;
+
+            let mut dapp_info =
+                IntegratedDApps::<T>::get(&smart_contract).ok_or(Error::<T>::ContractNotFound)?;
```

Comment:
I noticed in `stake` and `unstake` it returns `NotOperatedDApp` error instead. Maybe worth to unify them?